### DISCUSSION TITLE
Phone input component

### DISCUSSION THREAD

**Discussion by NemirStephan** [Legal-and-General/canopy#265] [Ideas]
I'm wondering if we should add a phone input component to Canopy that extends the existing `lgInput` when the type attribute is set to 'tel. The component could provide some default behaviour such as

- Restrict input to '+', numeric characters and spaces
- Automatically restricting the number of numeric characters to 16 digits 
- Allow '+' only as the first character

**Comment by andy-polhill** [comment]
I'm not convinced about the criteria for this one, I might be wrong but forcing people to enter phone numbers with the full international code and '+' prefix doesn't feel like a great user experience. It might be that I need more of a pitch from the UX team 😁

**Reply by NemirStephan** [reply]
I don't think that we want to force them to enter an international phone number, if the phone number starts with a leading single zero for example, then we can assume it's a UK number. In our case we want to verify the number with a back end service so we need to ensure the numbers are in some standard format. For example, these would be considered valid number formats.

- 01273 123 456
- +441273 123 456
- 00441273 123 456

So I think the idea is to have a consistent prompt to encourage users to enter an international format with a hint and to pre populate empty fields with '+44'

**Reply by NemirStephan** [reply]
Rather than adding a new input component, we can use a standard input of type=tel and further restrict input with a pattern attribute. For example,
```html
<input 
  type="tel" id="phone"
  pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}">
```

We still need to figure out the correct pattern to allow

- An optional leading '+'
- Numbers and spaces only

Again, it might be worth adding an example to the Canopy storybook

**Reply by elenagarrone** [reply]
> I don't think that we want to force them to enter an international phone number, if the phone number starts with a leading single zero for example, then we can assume it's a UK number. In our case we want to verify the number with a back end service so we need to ensure the numbers are in some standard format. For example, these would be considered valid number formats.
> 
> * 01273 123 456
> * +441273 123 456
> * 00441273 123 456
> 
> So I think the idea is to have a consistent prompt to encourage users to enter an international format with a hint and to pre populate empty fields with '+44'

I don't think you can assume that if it starts with a 0 then it's a UK number :/ in italy all the phone numbers starts with 0 🤔

**Reply by NemirStephan** [reply]
I think thats probably true for most countries. But If the user enters a number like 01273 123 456 then we will not be able to validate it with a backend service such as Loqate unless we supply the country code. So we have 2 options:-

1. Force the user to enter an international number
2. Assume a default country code such as GB if they have entered something like 01273 123 456
